### ISSUE TITLE
Any way to keep local collection JSON and what is in Postman in sync?

### ISSUE THREAD

**Issue by bunchjesse** [a85/POSTMan-Chrome-Extension#3475]
Hi there,

I have a collection and I've exported it as JSON so that I could run it in CI with newman.

Now, if I make changes to the collection in Postman, what is the best way to keep the JSON in sync? Ideally, I'd like to avoid having issues where what is in our repo is different from what we have in Postman.

Just thinking outside the box a little, could newman just download a fresh JSON file at the start of the test run? Is that possible?

**Comment by djbf** [comment]
I'd suggest regarding Postman as an IDE and storing your exported JSON file under version control. Your CI process could then check out the latest version of your JSON file prior to calling the Newman task. Newman can also run a collection that's accessible via URL, if that helps at all.

**Comment by bunchjesse** [comment]
Thanks, @djbf.

> I'd suggest regarding Postman as an IDE and storing your exported JSON file under version control. 

This is exactly what I'm doing. The problem is that I have to *remember* to re-export the collection every time I make a change to it. What is stored in Postman does not stay in sync with the JSON file.

> Newman can also run a collection that's accessible via URL, if that helps at all.

This would be a great option if not for the detail that the link is not automatically updated with changes you make in Postman. You'd have to generate a new link each time (from my reading of the warning in Postman).

**Comment by sdnts** [comment]
@bunchjesse @djbf You can also use Sync. This was precisely why we built it :)

Newman can also [run collections from a URL](https://github.com/postmanlabs/newman#getting-started), so what you can do, is use the Postman Pro API to get the collection link (this will be done only once), and supply that to Newman. Now, anytime you change your collection in the app, Sync will update the collection. When Newman runs, it will fetch the latest version (since the URL remains the same).

I'm closing this, but please let me know if I misunderstood, so I can reopen.

**Comment by mohit-sentieo** [comment]
@madebysid  One problem with using sync directly is that we are not able to review changes to the collection. The normal github pull request -> code review flow works great with this. Would be great if there is even a sync button where postman can resync from a specific location on disk.

**Comment by tyomo4ka** [comment]
@madebysid I'm not sure if we need to reopen this or create a new issue. Actually, we are having an issue with the way how the share collection functionality works.

When I generate a URL using share collection feature  this URL works only for me. If another person is changing the collection his/her changes are not picked up under this URL.

This is super annoying as it doesn't really allow us to collaborate on the collection and we have one man dependency.
